openpype/hosts/substancepainter/api/lib.py

```diff
@@ -588,6 +588,7 @@ def _setup_file_dialog():
         # the file
         while not file_dialog.selectedFiles():
             app.processEvents(QtCore.QEventLoop.ExcludeUserInputEvents, 1000)
+            continue
```

Comment:
Sorry, I don't see how this fixes anything - Python will just continue the `while` loop as usual. Pretty sure this is not a fix for whatever issue you might be facing.

Reply:
The issue we are really facing here is that for whatever reason this https://github.com/ynput/OpenPype/pull/6004/files#diff-89e89807d0d0ec24aecd35b91cb64013cf39e14013f4963d802a8a384888be28R585 call to `file_dialog.selectUrl` never ends up setting the selected file. The resaon _could be_ because the file does not exist? However, that should be asserted higher up so the file should exist.

As such, the issue lies in the fact that the widget is not returning the file is selected.

Reply:
We could also take out the process events from the while loop and make the timeout larger, e.g. `5000` to capture MOST use cases? But it might fail for those where it doesn't finish setting the selected file in the UI.

Reply:
We could also avoid this hacky "widget hacking" completely if we were to just write our own full Qt replacement for whatever is the "New file dialog" in substance painter but preferably we can stay as native to theirs as possible.

Reply:
> We could also take out the process events from the while loop and make the timeout larger, e.g. `5000` to capture MOST use cases? But it might fail for those where it doesn't finish setting the selected file in the UI.

Maybe we can try this first to see if the widget is returning to set up the selected files.

Reply:
> if the widget is returning to set up the selected files.

Haha, well it clearly isn't - otherwise it'd exit the while loop 🗡️